crates/biome_js_analyze/src/lint/nursery/no_yoda_expression.rs

```diff
@@ -0,0 +1,430 @@
+use crate::{utils::is_node_equal, JsRuleAction};
+use biome_analyze::{
+    context::RuleContext, declare_rule, ActionCategory, Ast, FixKind, Rule, RuleDiagnostic,
+    RuleSource,
+};
+use biome_console::markup;
+use biome_diagnostics::Applicability;
+use biome_js_factory::make::{self, js_binary_expression, token};
+use biome_js_syntax::{
+    AnyJsExpression, AnyJsLiteralExpression, AnyJsStatement, JsBinaryExpression, JsBinaryOperator,
+    JsCallExpression, JsLanguage, JsLogicalExpression, JsLogicalOperator, JsSyntaxKind,
+    JsUnaryOperator, JsYieldArgument, JsYieldExpression, T,
+};
+use biome_rowan::{
+    AstNode, BatchMutationExt, NodeOrToken, SyntaxTriviaPiece, TriviaPieceKind, WalkEvent,
+};
+
+declare_rule! {
+    /// Disallow the use of yoda expressions.
+    ///
+    /// A Yoda expression is a programming style where the two parts of the expression are reversed from the typical order in a conditional statement.
```

Comment:
```suggestion
    /// A Yoda expression is a programming style where, given a binary operation, the "static" part of the binary operation is placed on the left-hand side.
```

I suggested a different wording. Using the text from wikipidia isn't helpful because it uses very technical words. We try to use "plain" terms, so beginners can understand: https://biomejs.dev/internals/philosophy/#technical

> Reduce jargon. Don’t assume that users will understand specific terminology. Strive to provide precise meaning for experts and beginners. For example, use “character” where you would traditionally use “token” when producing parser errors.